src/lib/object-diff.js

```diff
@@ -0,0 +1,27 @@
+var assert = require('assert')
+
+function difference(a, b) {
+  var diff = {}
+
+  Object.keys(a).forEach(compare(diff, a, b))
+  Object.keys(b).forEach(compare(diff, a, b))
+
+  return diff
+}
+
+function compare(diff, a, b) {
+  return function curried(n) {
+    if (!diff[n]) {
+      try {
+        assert.deepEqual(a[n], b[n])
+      } catch (e) {
+        diff[n] = {
+          config1: a[n],
+          config2: b[n],
```

Comment:
Should we prune the `a[n]` and `b[n]` from `a` and `b` respectively, after this step? Or at least `b[n]` from `b`, otherwise  we will be stepping over the config which has already been considered again for
```
Object.keys(b).forEach(compare(diff, a, b))
```